src/storage/conversion/ContentTypeReplacer.ts

```diff
@@ -0,0 +1,62 @@
+import type { Representation } from '../../ldp/representation/Representation';
+import { RepresentationMetadata } from '../../ldp/representation/RepresentationMetadata';
+import type { ValuePreferences } from '../../ldp/representation/RepresentationPreferences';
+import { CONTENT_TYPE } from '../../util/Vocabularies';
+import { matchingMediaTypes } from './ConversionUtil';
+import type { RepresentationConverterArgs } from './RepresentationConverter';
+import { RepresentationConverter } from './RepresentationConverter';
+
+/**
+ * A {@link RepresentationConverter} that changes the content type
+ * but does not alter the representation.
+ *
+ * Useful for when a content type is binary-compatible with another one;
+ * for instance, all JSON-LD files are valid JSON files.
+ */
+export class ContentTypeReplacer extends RepresentationConverter {
+  private readonly replacements: Record<string, ValuePreferences> = {};
+
+  /**
+   * @param replacements - Map of content types and content types to replace them by.
+   */
+  public constructor(replacements: Record<string, string>);
+  public constructor(replacements: Record<string, Iterable<string>>);
+  public constructor(replacements: Record<string, any>) {
```

Comment:
```suggestion
  public constructor(replacements: Record<string, string | Iterable<string>>) {
```
I would use this interface since that is what is actually supported by the constructor (unless you're not a fan of a Record potentially containing multiple types 😄 ).

Reply:
Components.js doesn't take it.

Reply:
Ah yea, forgot about that issue. Nevermind then.